core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -408,4 +408,35 @@ class DeleteTopicTest extends ZooKeeperTestHarness {
     val leaderIdOpt = zkClient.getLeaderForPartition(new TopicPartition(topic, 0))
     assertTrue("Leader should exist for topic test", leaderIdOpt.isDefined)
   }
+
+  @Test
+  def testDeletingPartiallyDeletedTopic() {
+    /**
+      * A previous controller could have deleted some partitions of a topic from ZK, but not all partitions, and then crashed.
+      * In that case, the new controller should be able to handle the partially deleted topic, and finish the deletion.
+      */
+
+    val replicaAssignment = Map(0 -> List(0, 1, 2), 1 -> List(0, 1, 2))
+    val topic = "test"
+    servers = createTestTopicAndCluster(topic, true, replicaAssignment)
+
+    /**
+      * shutdown all brokers in order to create a partially deleted topic on ZK
+      */
+    servers.foreach(_.shutdown())
```

Comment:
Is the remaining partition guaranteed to have no leader after this point? By default, unclean leader election is disabled.

Reply:
To guarantee that partition 0 will not have leadership info after the servers come back later at line 438, I used the zkClient to explicitly delete the znode /brokers/topics/test/partition/0 at line 432.
Does that answer your question?